core/os/file_access.cpp

```diff
@@ -252,10 +252,10 @@ String FileAccess::get_token() const {
 
 class CharBuffer {
 	Vector<char> vector;
-	char stack_buffer[256];
+	char stack_buffer[256]; // It does not need to be initialized with zeroes, because it is always used in the correct way.
```

Comment:
Is there a special comment you can use to disable the cppcheck warning, so that it's still resolved?

Reply:
This does not suppress Cppcheck's warning, but it does inform why this board is not initiated with zeros.
I wrote it thinking it's a single case, but at the moment I found already ~10 such large arrays, so it makes no sense to describe the same thing with each one, so I also removed this comment.

Reply:
Yes we discussed it briefly on IRC and the general consensus is that buffers should always stay uninitialized: https://freenode.logbot.info/godotengine-devel/20201123#c5936108-c5937054

I think it could be useful to have a standard comment we can put in such case to make things explicit.

And most linters have specific comments that can be used to turn off warnings on a specific line, maybe cppcheck has that too? If so we could just use that.

Reply:
Seems possible as per https://sourceforge.net/p/cppcheck/discussion/general/thread/361d4f51/

Reply:
Looks that suppression of `uninitMemberVar` doesn't work.
I tried to put it comment where variable is created and also in constructor, but none of this worked.

Reply:
Did you use `--inline-suppr` on the command line as suggested in that thread?

Reply:
I tried to use cppcheck with and without this command, but there wasn't any difference

Reply:
I tested and it seems to work, but the suppress comment needs to be on the line which triggers it, which is sadly not the variable declaration, but the constructor. This works to suppress two warnings:
```cpp
class Test {
public:
	int a;

	Test() {} // cppcheck-suppress uninitMemberVar
};

int main() {
	Test t = Test(); // cppcheck-suppress unreadVariable
	return 0;
}
```
```
$ cppcheck --enable=all test.cpp 
Checking test.cpp ...
test.cpp:5:2: warning: Member variable 'Test::a' is not initialized in the constructor. [uninitMemberVar]
 Test() {} // cppcheck-suppress uninitMemberVar
 ^
test.cpp:9:9: style: Variable 't' is assigned a value that is never used. [unreadVariable]
 Test t = Test(); // cppcheck-suppress unreadVariable
        ^
$ cppcheck --enable=all --inline-suppr test.cpp 
Checking test.cpp ...
```

Reply:
But since it would silence the error for all member variables, I'm not sure we should use it.